demo_nodes_py/demo_nodes_py/parameters/async_param_client.py

```diff
@@ -0,0 +1,170 @@
+
+    node.get_logger().info('Listing Parameters:')
+    future = client.list_parameters(param_file, 10)
+    client_executor.spin_until_future_complete(future)
+    initial_parameters = future.result()
+    if initial_parameters:
+        for param_name in initial_parameters.result.names:
+            node.get_logger().info(f'    - {param_name}')
+    else:
+        node.get_logger().error(f'Error listing parameters: {future.exception()}')
+
+    node.get_logger().info('Getting parameters:')
+    future = client.get_parameters(param_file)
+    client_executor.spin_until_future_complete(future)
+    parameter_values = future.result()
+    if parameter_values:
+        for i, v in enumerate(parameter_values.values):
+            node.get_logger().info(f'    - {param_file[i]}: {parameter_value_to_python(v)}')
+    else:
+        node.get_logger().error(f'Error getting parameters: {future.exception()}')
+
+    node.get_logger().info('Setting parameters:')
+    # We can do this since client.set_parameters takes
+    # List[Union[rclpy.parameter.Parameter, rcl_interfaces.msg.Parameter]]
+    future = client.set_parameters([
+        Parameter('int_parameter', Parameter.Type.INTEGER, 10).to_parameter_msg(),
+        Parameter('string_parameter', Parameter.Type.STRING, 'Fee Fi Fo Fum'),
+        Parameter('bool_parameter', Parameter.Type.BOOL, True),
+    ])
+    client_executor.spin_until_future_complete(future)
+    parameter_values = future.result()
+    if parameter_values:
+        for i, v in enumerate(parameter_values.results):
+            node.get_logger().info(f'    {param_file[i]}:')
+            node.get_logger().info(f'        successful: {v.successful}')
+    else:
+        node.get_logger().error(f'Error setting parameters: {future.exception()}')
+
+    node.get_logger().info('Getting parameters:')
+    future = client.get_parameters(param_file)
+    client_executor.spin_until_future_complete(future)
+    parameter_values = future.result()
+    if parameter_values:
+        for i, v in enumerate(parameter_values.values):
+            node.get_logger().info(f'    {param_file[i]}: {parameter_value_to_python(v)}')
+    else:
+        node.get_logger().error(f'Error getting parameters: {future.exception()}')
+
+    node.get_logger().info('Describing parameters:')
+    future = client.describe_parameters(param_file)
+    client_executor.spin_until_future_complete(future)
+    parameter_descriptions = future.result()
+    if parameter_descriptions:
+        for i, v in enumerate(parameter_descriptions.descriptors):
+            node.get_logger().info(f'    {param_file[i]}:')
+            for s in v.__slots__:
+                node.get_logger().info(f'        {s}: {getattr(v, s)}')
+    else:
+        node.get_logger().error(f'Error describing parameters: {future.exception()}')
+
+    node.get_logger().info('Getting parameter types:')
+    future = client.get_parameter_types(param_file)
```

Comment:
Interesting, I didn't notice that we added a `get_parameter_types` method 🙃 
It is provided redundant information as `describe_parameters`, so maybe we can just omit it here for simplicity.